developer_manual/digging_deeper/groupware/calendar_provider.rst

```diff
@@ -0,0 +1,637 @@
+
+Again, there is a function ``hasCalendarInCalendarHome`` that checks if a certain combination of ``principalUri`` and ``calendarUri`` are existing. Here, it is just hard-coded to exactly one calendar.
+
+Finally, there is a function to query for a single calendar instance using ``getCalendarInCalendarHome``. It returns a single calendar instance or ``null`` if no matching calendar is found.
+
+Register the calender provider
+------------------------------
+
+As a last step, you must register the calendar provider in your ``info.xml`` by adding
+
+.. code-block:: xml
+
+    <sabre>
+        <calendar-plugins>
+            <plugin>OCA\YourAppName\DAV\CalendarPlugin</plugin>
+        </calendar-plugins>
+    </sabre>
+
+With all these steps done, you should be able to see the calender(s) in the calendar app and the CalDAV interface of the core.
+
+Appendix: Registering the calendar with the PHP API interface
+-------------------------------------------------------------
+
+Additionally to the registration in the DAV app, the core provides another way to register a calendar.
+
+.. note:: Currently, the PHP API is not used by the DAV app. Any registered calendar will not automatically show up in the calendar's view or the CalDAV list. This might change in the future, thus it might be a good idea to provide this interface as well.
+
+Read-only support
+~~~~~~~~~~~~~~~~~
+
+To provide calendar(s) you have to write a class that implements the ``OCP\Calendar\ICalendarProvider`` interface.
+
+.. code-block:: php
+
+    <?php
+
+    use OCP\Calendar\ICalendarProvider;
+
+    class CalendarProvider implements ICalendarProvider {
+
+        public function getCalendars(string $principalUri, array $calendarUris = []): array {
+            $calendars = [];
+            // TODO: Run app specific logic to find calendars that belong to
+            //       $principalUri and fill $calendars
+
+            // The provider can simple return an empty array if there is not
+            // a single calendar for the principal URI
+            if (empty($calendars)) {
+                return [];
+            }
+
+            // Return instances of \OCP\Calendar\ICalendar
+            return $calendars;
+        }
+    }
+
+This ``CalendarProvider`` class is then registered in the :ref:`register method of your Application class<Bootstrapping>` with ``$context->registerCalendarProvider(CalendarProvider::class);``.
+
+
+Write support
+~~~~~~~~~~~~~
+
+Calendars that only return `ICalendar` are implicitly read-only. If your app's calendars can be written to, you may implement the ``ICreateFromString``. It will allow other apps to write calendar objects to the calendar by passing the raw iCalendar data as string.
```

Comment:
```suggestion
Calendars that only return `ICalendar` are implicitly read-only. If your app's calendars can be written to, you may implement the ``ICreateFromString`` interface. It will allow other apps to write calendar objects to the calendar by passing the raw iCalendar data as string.
```